packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -177,6 +177,40 @@ String generateNumberFormattingLogic(Message message) {
   return formatStatements.isEmpty ? '@(none)' : formatStatements.join();
 }
 
+/// To make it easier to parse plurals or select messages, temporarily replace
+/// each "{placeholder}" parameter with "#placeholder#" for example.
+String _replacePlaceholdersBraces(
+  String translationForMessage,
+  Iterable<Placeholder> placeholders,
+  String replacementBraces,
+) {
+  assert(replacementBraces.length == 2);
+  String easyMessage = translationForMessage;
+  for (final Placeholder placeholder in placeholders) {
+    easyMessage = easyMessage.replaceAll(
+        '{${placeholder.name}}', '${replacementBraces[0]}${placeholder.name}${replacementBraces[1]}');
+  }
+  return easyMessage;
+}
+
+String _replacePlaceholdersWithVariables(String message, Iterable<Placeholder> placeholders, [String braces = '{}']) {
```

Comment:
This is a nice bit of factoring. A comment that provided examples of the transformations it makes would be helpful.

Reply:
I have added a comment that describes what it does with 2 examples.